tests/unit/scheduler.c

```diff
@@ -336,6 +337,27 @@ int test_threshold_scheduler()
 	}
 	nccl_net_ofi_release_schedule(scheduler, schedule);
 
+	schedule = scheduler->get_schedule(scheduler, rr_threshold + 1, num_rails);
+	if (!schedule) {
+		NCCL_OFI_WARN("Failed to get schedule");
+		free(ref_schedule);
+		return -1;
+	}
+	ref_schedule->num_xfer_infos = 2;
+	ref_schedule->rail_xfer_infos[0].rail_id = 2;
+	ref_schedule->rail_xfer_infos[0].offset = 0;
+	ref_schedule->rail_xfer_infos[0].msg_size = rr_threshold / 2 + 128;
+	ref_schedule->rail_xfer_infos[1].rail_id = 3;
+	ref_schedule->rail_xfer_infos[1].offset = rr_threshold / 2 + 128;
+	ref_schedule->rail_xfer_infos[1].msg_size = rr_threshold / 2- 127;
+	ret = verify_schedule(schedule, ref_schedule);
+	if (ret) {
+		NCCL_OFI_WARN("Verification failed");
+		free(ref_schedule);
+		return ret;
+	}
+	nccl_net_ofi_release_schedule(scheduler, schedule);
+
 	/* Verify that three messages with `rr_threshold' bytes are assigned round robin */
```

Comment:
I think the reason we had this tested 3 times is to check the wrap-around case. Now you are not testing that case anymore, since you are doing this 4 times on 4 rails. And the comment is not correct anymore :)

I think it would be better to have a for loop for this instead of doing the same thing multiple times.

Reply:
I have modified the test-case to now test Round-robin on 1 rail, 2 rails, 3 rails and 4 rails. All the way not resetting the RR-Counter. This way we have covered the wrap-around case as well.